### DISCUSSION TITLE
Using "-virtual-pixel none" with "pattern:"

### DISCUSSION THREAD

**Discussion by GeeMack** [ImageMagick/ImageMagick#5331] [Development] [answered]
Using ImageMagick 7.1.0-43 Q16-HDRI x64 on Windows 10, I've come across an issue when using "-virtual-pixel none"  and "pattern:". I would expect this command to create a 240x240 image of a checkerboard pattern, black and white, single pixels. Instead it creates an all black image...

    magick -size 240x240 -virtual-pixel None pattern:gray50 test.png

![test](https://user-images.githubusercontent.com/65475279/179367751-ba22ce88-d190-45cf-addc-46d3c24fea31.png)

Using a virtual-pixel setting of "none" or "transparent" results in this black output. Using any other setting results in the "pattern:gray50" checkerboard image as expected...

    magick -size 240x240 -virtual-pixel Mirror pattern:gray50 test2.png

![test2](https://user-images.githubusercontent.com/65475279/179367800-6d95dc5f-e6e8-4b89-bd35-a7120559cbeb.png)

**Comment by snibgo** [comment]
Weird.

The problem occurs for other patterns, such as crosshatch30. It does not occur if there is no "-size" option.

**Comment by urban-warrior** [comment, marked as answer]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/archive/beta/ by sometime tomorrow.

**Comment by GeeMack** [comment]
I have now installed ImageMagick 7.1.0-44 Q16-HDRI x64, and this problem appears to be resolved. Thanks!
